source/elements/oneDAL/include/oneapi/dal/common.hpp

```diff
@@ -0,0 +1,58 @@
+namespace oneapi::dal {
+
+class base {
+public:
+    virtual ~base() = default;
+};
+
+enum class data_type {
```

Comment:
do we want to specify the base scalar type explicitly?

Reply:
For now, I do not see a use-case for that...